Add Entra support article (#1590)
dnsimple/dnsimple-support: content/articles/entra-identity-provider.md

---
title: Entra as an Identity Provider
excerpt: How to use Entra to create and log in to accounts.
meta: Learn how to seamlessly integrate Entra as your identity provider for account creation and secure logins, enhancing your user experience and security.
categories:
- Account
---

# Entra as an Identity Provider

### Table of Contents {#toc}

* TOC
{:toc}

---

Using Entra as an identity provider for single sign-on streamlines the login experience for you and your team and helps you manage risk at scale.

<info>
Entra as an Identity Provider is only available on Enterprise plans. See our [pricing page](https://dnsimple.com/pricing) for more details.
</info>

## Prerequisites

To proceed with configuring login with SSO through Entra, you must:

- Have administrator access to a DNSimple account.
- Be subscribed to a DNSimple Enterprise plan.

## Linking an Entra organization to a DNSimple account

DNSimple's Entra app integration is pending review and not yet available for installation via the [Microsoft Entra App Gallery](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-integrated-apps).

### Creating a customer Entra app

> For now, you can use Entra as an Identity Provider by [creating your own custom <label>Single tenant</label> Entra app integration](https://docs.azure.cn/en-us/data-explorer/kusto/access-control/provision-entra-id-app?view=microsoft-fabric&tabs=portal) to work with DNSimple by following these steps. Don't forget to take note of your <label>Directory (tenant) ID</label>, <label>Application (client) ID</label>, and <label>Client secret</label>.

Reviewer: I noticed our `label`s don't have any styling. I'd suggest removing them, but we have a lot of them across all articles.

Reviewer 2: I've been updating these to ** ** when I come across them, hopefully will catch them all with this revamp.

### Access control

You can set the level of DNSimple access for each member under the <label>Members and seats</label> tab in your DNSimple Account settings. For full details, visit [Domain Access Control](https://support.dnsimple.com/articles/domain-access-control/).

## Linking a DNSimple user to an Entra identity

When a DNSimple user exists in a DNSimple account before SSO is enabled, they will be required to link their Entra identity to verify ownership of both DNSimple and Entra user accounts. This is done on your DNSimple <label>User settings</label> page.
To link a DNSimple user to an Entra identity:
1. At DNSimple, use the account switcher at the top right of the page to go to your User Settings page.
1. Scroll down to the **Identities** card, and click **Add** next to the Entra identity provider.
1. Authenticate your Entra account.
1. You can now log in to DNSimple using the linked Entra identity.

## Logging in via Entra

You'll need to [link an Entra organization to your DNSimple account](#linking-an-entra-organization-to-a-dnsimple-account) before your team members can log in via Entra SSO.

> 1. To log in to DNSimple using Entra, visit https://dnsimple.com/login.

Reviewer: ```suggestion
1. To log in to DNSimple using Entra, visit [https://dnsimple.com/login](https://dnsimple.com/login).
```

1. Click Sign in using Entra.
1. Enter the organization Entra domain or tenant ID and click Sign in.
1. If you are prompted for your Entra username and password, enter them.
1. If your credentials are valid, you will be redirected back to DNSimple and logged in.

<info>

> If you are not yet a member of the DNSimple account, the account administrator will receive a notification to grant access. Once they have granted access, you will be able to see the DNSimple account's assets.

Author: It is true, there's emails sent when someone joins to the same org.

> If you are not yet a member of the DNSimple account, the account administrator will receive a notification to grant access. Once they have granted access, you will be able to see the DNSimple account's assets.

Author: And unless I missed something, the suggested copy is the same we have in place 😕

> If you are not yet a member of the DNSimple account, the account administrator will receive a notification to grant access. Once they have granted access, you will be able to see the DNSimple account's assets.

Reviewer: LOL – you're right – its the same. I'm not sure what I was seeing here. 🙈

If you log out of Entra, you will also be logged out of DNSimple.
</info>

## Unlinking an Entra organization from a DNSimple account

1. At DNSimple, use the account switcher at the top right of the page to select the Account you want to unlink. 

> 2. Once you have selected the account, go back to the account switcher and go to Account Settings. 

Reviewer: > go back to the account switcher

😭

3. On the left-hand side, click on the Single-Sign On tab. 
4. Click Revoke next to the Entra SSO provider to remove the link from DNSimple to Entra.

## Unlinking a DNSimple user from an Entra identity

> 1. At DNSimple, use the account switcher at the top right of the page to go to your User Settings page.

Reviewer: You will be met with an error if you try to do this because we don't allow it.

I get the feeling we should remove the two sections on this page relating to User Identity. It's just too confusing. Users just log in – nothing required to "link"/setup or "unlink"/remove.

It is true that you can see which identities are linked on the User Settings page, but I'm not sure how noteworthy that is.

- Linking a DNSimple user to an Entra identity
- Unlinking a DNSimple user from an Entra identity

Reviewer: Google SSO is in the same boat as Okta and Entra.

This feature IS relevant for Google via social.

categories:
- Account
---

# Entra as an Identity Provider

### Table of Contents {#toc}

* TOC
{:toc}

---

Using Entra as an identity provider for single sign-on streamlines the login experience for you and your team and helps you manage risk at scale.

<info>
Entra as an Identity Provider is only available on Enterprise plans. See our [pricing page](https://dnsimple.com/pricing) for more details.
</info>

## Prerequisites

To proceed with configuring login with SSO through Entra, you must:

- Have [Application Administrator](https://learn.microsoft.com/en-us/entra/identity/role-based-access-control/permissions-reference#application-administrator) access to a Microsoft Entra account.
- Have administrator access to a DNSimple account.
- Be subscribed to a DNSimple [Enterprise plan](https://dnsimple.com/enterprises).

## Linking an Entra organization to a DNSimple account

DNSimple's Entra app integration is pending review and is not yet available for installation via the [Microsoft Entra App Gallery](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-integrated-apps).

### Creating a customer Entra app

> For now, you can use Entra as an Identity Provider by [creating your own custom **Single tenant** Entra app integration](https://docs.azure.cn/en-us/data-explorer/kusto/access-control/provision-entra-id-app?view=microsoft-fabric&tabs=portal) to work with DNSimple by following these steps. Take note of your **Directory (tenant) ID**, **Application (client) ID**, and **Client secret**.

Reviewer 3: @dallasread, we are referencing the link above it, correct?

### Adding Redirect URIs

After creating your app, add the [Redirect URIs](https://learn.microsoft.com/en-us/entra/identity-platform/how-to-add-redirect-uri) for your new web app using the following URIs:
1. https://dnsimple.com/identity_providers/entra/callbacks/users/login
1. https://dnsimple.com/identity_providers/entra/callbacks/accounts/link
1. https://dnsimple.com/identity_providers/entra/callbacks/users/link

### Adding API permissions

> Add the following API Permissions to the app and grant access:
> 
> - openid
> - email
> - profile
> - User.Read
> - User.Read.All

Reviewer: ```suggestion
Add the following [API Permissions](https://learn.microsoft.com/en-us/entra/identity-platform/howto-update-permissions) to the app:

- openid
- email
- profile
- User.Read
- User.Read.All
```

You'll also need to [grant consent](https://learn.microsoft.com/en-us/entra/identity-platform/howto-update-permissions?pivots=portal#grant-consent-for-the-added-permissions-for-the-enterprise-application) for these permissions.

### Creating a customer Entra app

For now, you can use Entra as an Identity Provider by [creating your own custom **Single tenant** Entra app integration](https://docs.azure.cn/en-us/data-explorer/kusto/access-control/provision-entra-id-app) to work with DNSimple by following these steps. Take note of your **Directory (tenant) ID**, **Application (client) ID**, and **Client secret**.

### Adding Redirect URIs

After creating your app, add the [Redirect URIs](https://learn.microsoft.com/en-us/entra/identity-platform/how-to-add-redirect-uri) for your new web app using the following URIs:
1. https://dnsimple.com/identity_providers/entra/callbacks/users/login
1. https://dnsimple.com/identity_providers/entra/callbacks/accounts/link
1. https://dnsimple.com/identity_providers/entra/callbacks/users/link

### Adding API permissions

Add the following [API Permissions](https://learn.microsoft.com/en-us/entra/identity-platform/howto-update-permissions) to the app:

- `openid`
- `email`
- `profile`
- `User.Read`

> - `User.Read.All`

Author: ```suggestion
- `User.Read.All`
- `Domain.Read.All`
```
